src/Currency.php

```diff
@@ -144,16 +145,31 @@ public function getAmountAsFractionalUnits(): int
     }
 
     /**
+     * @todo Use method getDecorationType() when decorating the result
      * {@inheritdoc}
      */
-    public function format(string $decimalsSeparator = '.', string $thousandsSeparator = '', int $extraPrecision = 0): string
+    public function format(CurrencyFormatInterface $currencyFormat = null): string
     {
-        $nDecimals = $this->currencyType->getNumFractionalDigits() + $extraPrecision;
+        if (is_null($currencyFormat)) {
+            $currencyFormat = new CurrencyFormat();
+        }
+
+        $precision = $currencyFormat->getPrecision();
+        if (is_null($precision)) {
+            $precision = $this->currencyType->getNumFractionalDigits();
+        }
+        $nDecimals = $precision + $currencyFormat->getExtraPrecision();
```

Comment:
The idea behind this "strange" desired behavior is to allow complete flexibility to specify how many digits we want to handle. We can specify the absolute number of digits, or an extra number of digits over the "default" digits specified by the currency type.